teos/src/dbm.rs

```diff
@@ -1175,7 +1189,52 @@ mod tests {
         dbm.store_tracker(uuid, &tracker).unwrap();
 
         // We should get all the appointments matching our locator back except from the triggered one
-        assert_eq!(dbm.load_appointments(Some(locator)), appointments);
+        assert_eq!(dbm.load_appointments(Some((locator, None))), appointments);
+    }
+
+    #[test]
+    fn test_load_appointments_with_locator_and_user_id() {
+        let dbm = DBM::in_memory().unwrap();
+
+        // create two appointment maps for two userId
+        let mut user_id1_appointments = HashMap::new();
+        let mut user_id2_appointments = HashMap::new();
+
+        let dispute_tx = get_random_tx();
+        let dispute_txid = dispute_tx.txid();
+        let locator = Locator::new(dispute_txid);
+
+        // generate two user ids
+        let user_id1 = get_random_user_id();
+        let user_id2 = get_random_user_id();
+
+        let user = UserInfo::new(AVAILABLE_SLOTS, SUBSCRIPTION_START, SUBSCRIPTION_EXPIRY);
+        dbm.store_user(user_id1, &user).unwrap();
+        dbm.store_user(user_id2, &user).unwrap();
+
+        let (uuid, appointment) =
+            generate_dummy_appointment_with_user(user_id1, Some(&dispute_txid));
+        dbm.store_appointment(uuid, &appointment).unwrap();
+        // Store the appointment for the first user_id made using our dispute tx.
+        user_id1_appointments.insert(uuid, appointment);
+
+        let (uuid, appointment) =
+            generate_dummy_appointment_with_user(user_id2, Some(&dispute_txid));
+        dbm.store_appointment(uuid, &appointment).unwrap();
+        // Store the appointment for the second user_id made using our dispute tx.
+        user_id2_appointments.insert(uuid, appointment);
+
+        // Validate that the first user_id appointment map matches the fetched appointments.
+        assert_eq!(
+            dbm.load_appointments(Some((locator, Some(user_id1))),),
+            user_id1_appointments
+        );
+
+        // Validate that the second user_id appointment map matches the fetched appointments.
+        assert_eq!(
+            dbm.load_appointments(Some((locator, Some(user_id2)))),
+            user_id2_appointments
+        );
```

Comment:
I would go for a bigger test scenario, not just two appointments.
something like the test above: `test_load_appointments_with_locator`